### PR TITLE
Support MySQL Protocol (TCP only)

### PR DESCRIPTION
**Pull request by longjiquan**
issue: #22149 

- This PR let Milvus support MySql protocol in TCP layer, but socket is not supported yet.
- With this PR, the users can use the rich MySql tools to connect to Milvus.
- Besides this, this PR supports very few syntaxs of MySql, including simple retriving and simple count aggregation.
- This PR also defined the ast node framework and all necessary nodes.
- The syntax parser is based on the antlr and the antlr-based parser can translate the sql statements to the defined node.
- Based on the ast node framework, this PR provides a visitor pattern which can traverse the ast easily.
- This PR provides a very simple executor which translates sql plan node to exist RPC API.
- The simple executor only support count aggregation and simple retriving now.
- This PR provides a plugable sql factory which allow developers switch all the components easily.

Signed-off-by: longjiquan <jiquan.long@zilliz.com>

### CODE REVIEW

**Comment by longjiquan** [comment]
image

File: internal/mysqld/server.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/xelabs/go-mysqlstack/driver"
+	"github.com/xelabs/go-mysqlstack/xlog"
+	"go.uber.org/zap/zapcore"
+)
+
+type Server struct {
+	listener *driver.Listener
+}
+
+func (s *Server) Start() error {
```
Comment (czs007) [inline internal/mysqld/server.go:16]: It is recommended to add waitgroup and wait for the background coroutine in Start to exit when Close

File: internal/mysqld/parser/antlrparser/utils.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	"strings"
+
+	"github.com/antlr/antlr4/runtime/Go/antlr/v4"
+)
+
+func CaseInsensitiveEqual(s1, s2 string) bool {
+	return strings.ToLower(s1) == strings.ToLower(s2)
+}
+
+// GetOriginalText don't work.
```
Comment (czs007) [inline internal/mysqld/parser/antlrparser/utils.go:13]: do you figure out why ?

**Comment by czs007** [comment]
resolve conflicts and add more unit test

File: internal/mysqld/parser/antlrparser/utils.go (hunk 2 above)
Comment (longjiquan) [reply internal/mysqld/parser/antlrparser/utils.go:13]: just see https://github.com/antlr/antlr4/issues/1302 for reference.

**Pull request closed without merging**